docs/api.rst

```diff
@@ -10126,3 +10126,82 @@ Get associated locations
    :statuscode 200: Locations succesfully queried.
    :statuscode 409: User is not logged in. Check error message for details.
    :statuscode 500: Internal Rotki error
+
+Staking events
+==============
+
+.. http:get:: /api/(version)/staking/kraken
+
+   .. note::
+      This endpoint can also be queried asynchronously by using ``"async_query": true``
+
+   .. note::
+      This endpoint also accepts parameters as query arguments.
+
+   Doing a GET on this endpoint will return all staking events for the desired location. At the moment
+   the only valid location is kraken.
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      GET /api/1/staking/kraken HTTP/1.1
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+      {"from_timestamp": 1451606400, "to_timestamp": 1571663098, "only_cache": false}
+
+   :reqjson int limit: Optional. This signifies the limit of records to return as per the `sql spec <https://www.sqlite.org/lang_select.html#limitoffset>`__.
+   :reqjson int offset: This signifies the offset from which to start the return of records per the `sql spec <https://www.sqlite.org/lang_select.html#limitoffset>`__.
+   :reqjson string order_by_attribute: Optional. This is the attribute of the history by which to order the results. If none is given 'timestamp' is assumed. Valid values are: ['timestamp', 'location', 'amount'].
+   :reqjson bool ascending: Optional. False by default. Defines the order by which results are returned depending on the chosen order by attribute.
+   :reqjson int from_timestamp: The timestamp from which to query. Can be missing in which case we query from 0.
+   :reqjson int to_timestamp: The timestamp until which to query. Can be missing in which case we query until now.
+   :reqjson bool only_cache: Optional.If this is true then the equivalent exchange/location is not queried, but only what is already in the DB is returned.
+
+   **Example Response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/json
+
+      {
+          "result": [
+              {
+                  "event_type": "income",
+                  "asset": "ETH2",
+                  "timestamp": 16368645884914,
+                  "location": "kraken",
+                  "amount": "0.0000103220",
+                  "usd_value": "0"
+              },
+              {
+                  "event_type": "staking receive asset",
+                  "asset": "ETH2",
+                  "timestamp": 16367401989674,
+                  "location": "kraken",
+                  "amount": "0.0600000000",
+                  "usd_value": "0"
```

Comment:
It just occured to me that at the result of this call we want to have usd_values. Otherwise it's sort of useless. Do you think it would make sense to do a run of the "usd detection" logic for the stake events during this api call? Only run it ofc for stake events and only for stake events that don't already have a value.